transportable-udfs-trino-plugin/src/main/java/com/linkedin/transport/trino/TransportConnector.java

```diff
@@ -0,0 +1,119 @@
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.function.FunctionId;
+import io.trino.spi.function.FunctionProvider;
+import io.trino.spi.transaction.IsolationLevel;
+import java.io.File;
+import java.io.FileFilter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.bval.util.StringUtils;
+
+import static java.util.Objects.*;
+
+
+/**
+ * This class implements the interface of Connector from Trino SPI as a part of Trino plugin
+ * to load UDF classes in Trino server following the development guideline
+ * in https://trino.io/docs/current/develop/spi-overview.html
+ */
+public class TransportConnector implements Connector {
+
+  private static final Logger log = Logger.get(TransportConnector.class);
+  private static final String DEFAULT_TRANSPORT_UDF_REPO = "transport-udf-repo";
+  private static final FileFilter TRANSPORT_UDF_JAR_FILTER = (file) ->
+      file.isFile() && file.getName().endsWith(".jar") && !file.getName().startsWith("transportable-udfs-api")
+      && !file.getName().startsWith("transportable-udfs-trino")
+      && !file.getName().startsWith("transportable-udfs-type-system")
+      && !file.getName().startsWith("transportable-udfs-utils");
+
+  private final ConnectorMetadata connectorMetadata;
+  private final FunctionProvider functionProvider;
+
+  public TransportConnector(ConnectorMetadata connectorMetadata, FunctionProvider functionProvider) {
+    this.connectorMetadata = requireNonNull(connectorMetadata, "connector metadata is null");
+    this.functionProvider = requireNonNull(functionProvider, "function provider is null");
+  }
+
+  @Inject
+  public TransportConnector(TransportConfig config) {
+    ClassLoader classLoaderForFactory = TransportConnectorFactory.class.getClassLoader();
+    List<URL> jarUrlList = getUDFJarUrls(config);
+    log.info("The URLs of Transport UDF jars: " + jarUrlList);
+    TransportUDFClassLoader classLoaderForUdf = new TransportUDFClassLoader(classLoaderForFactory, jarUrlList);
+    ServiceLoader<StdUdfWrapper> serviceLoader = ServiceLoader.load(StdUdfWrapper.class, classLoaderForUdf);
+    List<StdUdfWrapper> stdUdfWrappers = ImmutableList.copyOf(serviceLoader);
+    Map<FunctionId, StdUdfWrapper> functions = new HashMap<>();
+    for (StdUdfWrapper wrapper : stdUdfWrappers) {
+      log.info("Loading Transport UDF class: " + wrapper.getFunctionMetadata().getFunctionId().toString());
+      functions.put(wrapper.getFunctionMetadata().getFunctionId(), wrapper);
```

Comment:
We should not allow UDFs to have the same `FunctionId`.  Can you use 
```
functionsBuilder = ImmutableMap<FunctionId, StdUDfWrapper>.builder()
     .put(wrapper.getFunctionMetadata().getFunctionId(), wrapper)
     ....

// After loop
functions = functionsBuilder.buildOrThrow();
``` 
`ImmutableMap.Builder#put` disallows duplicate key by default ([reference](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html#put(K,%20V)))